### PR TITLE
Add flo based design to design concept

### PR DESCRIPTION
**Pull request by lstreckeisen**

### CODE REVIEW

File: SA/design_concept/content/design/design_proposals.typ [hunk 1]
```diff
@@ -1,8 +1,28 @@
+#import "../../../style.typ": include_section
+
 = Design Proposals
-In the following, all considered designs for function & data type composition are documented.
+In the following, all considered designs for function composition as well as the overall tool design are documented.
+
+== Overall tool design
+VisualFP should work in an IDE-style. This means that there is a way to organize defined functions, an editor for function composition and means to execute functions.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_proposals.typ:7]:
```suggestion
VisualFP should work in an IDE-style. This means that there is:

- An editor to compose functions
- An editor for custom types
- A way to organize defined functions and types
- Mean to execute functions
```

File: SA/design_concept/content/design/design_proposals.typ [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+
+@visualfp_ide_mockup shows a mockup of how the VisualFP IDE could look like.
+The two main elements are the sidebar, which gives the user access to pre- and self-defined functions, and the large editor-section.
+The designs proposed later in this section describe how the editor-section of the IDE could work.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_proposals.typ:11]:
```suggestion
The designs proposed later in this section describe how the function editor-section of the IDE could work.
```

File: SA/design_concept/content/design/design_proposals.typ [hunk 3]
```diff
@@ -1,8 +1,28 @@
...
+
+Functions in the sidebar can be grouped in modules, which appear like folders in the IDE.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_proposals.typ:13]:
```suggestion
Functions in the sidebar can be grouped in modules (analogous to Haskell modules), which appear like folders in the IDE.
```

File: SA/design_concept/content/design/design_proposals.typ [hunk 4]
```diff
@@ -1,8 +1,28 @@
...
+
+Regarding code execution, there are multiple considered options:
+- The defined functions are loaded into an interactive GHCi session where the user can interact with their functions.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_proposals.typ:16]:
```suggestion
- The defined content is loaded into an interactive GHCi session where the user can interact with it.
```

File: SA/design_concept/content/design/flo_inspired_design.typ [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+The types of a function or variable block can be viewed by hovering over it.
+
+An example of a simple function definition can be seen in @flo_inspired_addition, which is a visual representation of the addition code scenario defined in @design_eval_code_scenarios.
+
+#figure(
+    image("../../static/flo-inspired-addition.png", width: 50%),
+    caption: "Example of flow-inspired function definition for addition function"
+)<flo_inspired_addition>
+
+Functions themselves can of course also be used as parameters themselves.
```
Comment (fl3pp) [inline SA/design_concept/content/design/flo_inspired_design.typ:37]:
```suggestion
Functions can of course also be used as parameters themselves.
```

File: SA/design_concept/content/design/flo_inspired_design.typ [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+For that, the use can create a connector-arrow between the main function block (not one of the depending ones) and a function parameter slot in the same ways as with function parameters.
```
Comment (fl3pp) [inline SA/design_concept/content/design/flo_inspired_design.typ:38]:
```suggestion
For that, the user can create a connector-arrow between a function block and a function parameter slot in the same ways as with function parameters.
```

A function can have multiple parameters, thus a depending function can be used as well

File: SA/design_concept/content/design/flo_inspired_design.typ [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+When using functions as parameters, it is possible to auto-fill some of the functions parameters.
```
Comment (fl3pp) [inline SA/design_concept/content/design/flo_inspired_design.typ:39]: Do you mean left unapplied? 'auto-fill' suggests that the tooling is able to fill out parameters based on default values

File: SA/design_concept/content/design/flo_inspired_design.typ [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+An example for that can be seen in @flo_inspired_mapAddFive, which is a visual representation of the "Map Add 5" code scenario in @design_eval_code_scenarios.
+The auto-filled parameter of the "(+)" function can be recognized by the fill-color of the parameter slot.
+#figure(
+    image("../../static/flo-inspired-mapAdd5.png", width: 60%),
+    caption: "Example of functions used as parameters for other functions"
+)<flo_inspired_mapAddFive>
+
+Another important concept is recursion.
+In Haskell, recursive functions are often defined with pattern matching.
```
Comment (fl3pp) [inline SA/design_concept/content/design/flo_inspired_design.typ:50]: Recursion and pattern-matching are unrelated, I would split these two features

File: SA/design_concept/content/design/flo_inspired_design.typ (hunk 7 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/flo_inspired_design.typ:39]: Yes, I just couldn't think of another word

**Review by fl3pp — APPROVED** [review]

**Pull request merged**